dot/core/service.go

```diff
@@ -89,7 +90,14 @@ func NewService(cfg *Config) (*Service, error) {
 		return nil, ErrNilStorageState
 	}
 
-	if cfg.Network == nil {
+	var isNilNetwork bool
+	switch cfg.Network.(type) {
+	case Network:
+		isNilNetwork = cfg.Network == (*network.Service)(nil)
```

Comment:
Shouldn't this be

```suggestion
	case *network.Service:
		isNilNetwork = cfg.Network == (*network.Service)(nil)
```

? If there are other implementations of the `Network` interface, I think these should be added here.

Reply:
Updated.